farmerbot/internal/types.go

```diff
@@ -0,0 +1,194 @@
+package internal
+
+import (
+	"fmt"
+	"math"
+	"strings"
+	"time"
+
+	substrate "github.com/threefoldtech/tfchain/clients/tfchain-client-go"
+	zos "github.com/threefoldtech/zos/client"
+	"github.com/threefoldtech/zos/pkg"
+	"github.com/threefoldtech/zos/pkg/gridtypes"
+)
+
+// Config is the inputs for configuration for farmerbot
+type Config struct {
+	FarmID             uint32   `yaml:"farm_id"`
+	IncludedNodes      []uint32 `yaml:"included_nodes"`
+	ExcludedNodes      []uint32 `yaml:"excluded_nodes"`
+	NeverShutDownNodes []uint32 `yaml:"never_shutdown_nodes"`
+	Power              power    `yaml:"power"`
+}
+
+type powerState uint8
+
+const (
+	on = powerState(iota)
+	wakingUP
+	off
```

Comment:
why shuttingDown in order is after off? or is that how it's coded on chain? because the order isn't intuitive

Reply:
The order doesn't reflect anything